server/controllers/list.js

```diff
@@ -1104,6 +1104,10 @@ const addViewer = (req, resp) => {
     .then(() => {
       if (user) {
         resp.send(userToSend);
+        // eslint-disable-next-line no-param-reassign
+        resp.locales = userToSend;
```

Comment:
(This is a proposal only I don't know which approach is better, we have to discuss this):
We can do this in the same way as in activities case. Instead of adding middleware to every route we can call methods eg. sendAddViewerEmailNotification(params) in the controller. We can call it in fireAndForgetHelper also.

Reply:
I think it is a matter of choice. 
I think using middleware is clean and elegant

Reply:
Personally, I would go for a single dispatch and handle notification as an effect of action - this way there is a single `dispatch(ActivityType.LIST_ADD_USER, payload)` call that broadcasts the event over sockets, handles DB writes and optionally sends notifications.
The best option but hardest to implement as you do not store raw events nor change authors is to watch the DB for changes, aggregate them and then diff them to get change lists to be sent as notifications.